diff --git a/gcc/config/aarch64/darwin.h b/gcc/config/aarch64/darwin.h
index 5609c569dc1..9fcbfa3d777 100644
--- a/gcc/config/aarch64/darwin.h
+++ b/gcc/config/aarch64/darwin.h
@@ -65,10 +65,6 @@ along with GCC; see the file COPYING3.  If not see
 #define DARWIN_NOPIE_SPEC \
 " %<no-pie %<fno-pie %<fno-PIE "
 
-/* Hack alert - we want the exported cas etc.  */
-#undef LIB_SPEC
-#define LIB_SPEC "%{!static:-lSystem} -lgcc"
-
 /* Force the default endianness and ABI flags onto the command line
    in order to make the other specs easier to write.  Match clang in
    silently ignoring mdynamic-no-pic */
diff --git a/libgfortran/libgfortran.spec.in b/libgfortran/libgfortran.spec.in
index 367d485c230..058ee5775bf 100644
--- a/libgfortran/libgfortran.spec.in
+++ b/libgfortran/libgfortran.spec.in
@@ -5,4 +5,4 @@
 #
 
 %rename lib liborig
-*lib: @LIBQUADSPEC@ @LIBM@ %(libgcc) %(liborig)
+*lib: @LIBQUADSPEC@ @LIBM@ %(liborig)
