commit 23b0cf4402573babb7c3292a9f9d191cda1ee56b
Author: Matthias Koeppe <mkoeppe@math.ucdavis.edu>
Date:   Wed Jun 21 12:04:53 2023 -0700

    test/CMakeLists.txt (cmr_gtest): Explicitly link against gtest

diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 0bde995..7027ce5 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -37,7 +37,7 @@ endif()
 
 # Configure cmr_gtest target.
 target_compile_features(cmr_gtest PRIVATE cxx_auto_type)
-target_link_libraries(cmr_gtest gtest_main CMR::cmr)
+target_link_libraries(cmr_gtest gtest_main gtest CMR::cmr)
    
 include(GoogleTest)
 gtest_discover_tests(cmr_gtest)
