diff --git a/unix/Make-arch b/unix/Make-arch
index 08afb85..dbeb691 100644
--- a/unix/Make-arch
+++ b/unix/Make-arch
@@ -1471,6 +1471,25 @@
 	"RANLIB = ranlib" \
 	"LIBS = -L. -ltachyon $(MISCLIB) -lm -lpthread"

+# Linux Arm using gcc
+linux-arm:
+	$(MAKE) all \
+	"ARCH = linux-arm" \
+	"CFLAGS = -Wall -O3 -fomit-frame-pointer -ffast-math -DLinux $(MISCFLAGS)" \
+	"ARFLAGS = r" \
+	"STRIP = strip" \
+	"RANLIB = ranlib" \
+	"LIBS = -L. -ltachyon $(MISCLIB) -lm -lpthread"
+
+# Linux Arm using gcc, with threads
+linux-arm-thr:
+	$(MAKE) all \
+	"ARCH = linux-arm-thr" \
+	"CFLAGS = -Wall -O3 -fomit-frame-pointer -ffast-math -DLinux -DTHR -D_REENTRANT $(MISCFLAGS)" \
+	"ARFLAGS = r" \
+	"STRIP = strip" \
+	"LIBS = -L. -ltachyon $(MISCLIB) -lm -lpthread"
+
 # Linux x86 using gcc, threads, and OpenGL
 linux-thr-ogl:
 	$(MAKE) all \
