--- a/support/shobj-conf
+++ b/support/shobj-conf
@@ -127,8 +127,10 @@
 	SHOBJ_CFLAGS=-fPIC
 	SHOBJ_LD='${CC}'
 	SHOBJ_LDFLAGS='-shared -Wl,-soname,$@'
+	SHOBJ_LIBS='-Wl,--no-as-needed -lncursesw -Wl,--as-needed'
 
 	SHLIB_XLDFLAGS='-Wl,-rpath,$(libdir) -Wl,-soname,`basename $@ $(SHLIB_MINOR)`'
+	SHLIB_LIBS='-Wl,--no-as-needed -lncursesw -Wl,--as-needed'
 	SHLIB_LIBVERSION='$(SHLIB_LIBSUFF).$(SHLIB_MAJOR)$(SHLIB_MINOR)'
 	;;
 
@@ -171,7 +173,7 @@
 		;;
 	esac
 
-	SHLIB_LIBS='-lncurses'	# see if -lcurses works on MacOS X 10.1 
+	SHLIB_LIBS='-lncursesw'	# see if -lcurses works on MacOS X 10.1 
 	;;
 
 openbsd*|netbsd*|mirbsd*)
