Description: Adapt to GCC 6
Origin: https://src.fedoraproject.org/rpms/qepcad-B/blob/91dd25b/f/qepcad-B-gcc6.patch
Forwarded: https://github.com/chriswestbrown/qepcad/pull/2
Last-Update: 2021-06-23

--- a/plot2d/plot.cc
+++ b/plot2d/plot.cc
@@ -276,7 +276,7 @@
   V.resize(N);
   for(int i = 0; i < N; i++)
     in >> V[i];
-  return in;
+  return in.good();
 }
   
 void SNoverSR::glRend(const CADColors &C)
@@ -333,7 +333,7 @@
     }
   }
 
-  return in;
+  return in.good();
 }
   
 void SRoverSR::glRend(const CADColors &C)
@@ -351,7 +351,7 @@
 {
   in >> colorType >> a >> b.y;
   b.x = a.x;
-  return in;
+  return in.good();
 }
 
 void SRoverSN::glRend(const CADColors &C)
@@ -365,7 +365,8 @@
 
 bool SNoverSN::read(istream &in)
 {
-  return in >> colorType >> a;
+  in >> colorType >> a;
+  return in.good();
 }
   
 void SNoverSN::glRend(const CADColors &C)
@@ -380,7 +381,8 @@
 
 bool SN::read(istream &in)
 {
-  return in >> colorType >> x;
+  in >> colorType >> x;
+  return in.good();
 }
   
 void SN::glRend(const CADColors &C)
@@ -394,7 +396,8 @@
 
 bool SR::read(istream &in)
 {
-  return in >> colorType >> x1 >> x2;
+  in >> colorType >> x1 >> x2;
+  return in.good();
 }
   
 void SR::glRend(const CADColors &C)
