Description: Fix syntactically incorrect expressions.
Origin: https://src.fedoraproject.org/rpms/qepcad-B/blob/91dd25b/f/qepcad-B-syntax.patch
Forwarded: https://github.com/chriswestbrown/qepcad/pull/2
Last-Update: 2021-06-23

--- a/source/main/BEGINQEPCAD.c
+++ b/source/main/BEGINQEPCAD.c
@@ -53,7 +53,7 @@
   {
     int tmp = system("bash -c 'exit $(stty size | cut -d\" \" -f2)'");
     tmp = WEXITSTATUS(tmp);
-    if (10 <= tmp <= 512)
+    if (10 <= tmp && tmp <= 512)
       cols = tmp;
   }
 
--- a/source/userint/USERINT.c
+++ b/source/userint/USERINT.c
@@ -507,7 +507,7 @@
   Word L, S, T, CB;
 
   /* GET 1D CAD STACK */
-  L = L = LELTI(D,CHILD);
+  L = LELTI(D,CHILD);
   if (L == NIL)
   {
     SWRITE("Must by a 2D CAD!\n");
