From 66459b1b69fd90e1c9d620a934d4e3837ecad902 Mon Sep 17 00:00:00 2001
From: Dima Pasechnik <dima@pasechnik.info>
Date: Thu, 11 Jun 2026 22:38:19 -0500
Subject: [PATCH] add cmake build

---
 CMakeLists.txt | 37 +++++++++++++++++++++++++++++++++++++
 1 file changed, 37 insertions(+)
 create mode 100644 CMakeLists.txt

diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..d86b860
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,37 @@
+cmake_minimum_required(VERSION 3.30)
+project(Mcqd VERSION 1.0
+                  DESCRIPTION "finding a maximum clique in an undirected graph"
+                  LANGUAGES CXX)
+### Require out-of-source builds
+file(TO_CMAKE_PATH "${PROJECT_BINARY_DIR}/CMakeLists.txt" LOC_PATH)
+if(EXISTS "${LOC_PATH}")
+    message(FATAL_ERROR "You cannot build in a source directory")
+endif()
+
+add_executable(mcqdexe mcqd.cpp mcqd.h)
+add_library(mcqd SHARED mcqd.cpp mcqd.h)
+set_target_properties(mcqd PROPERTIES PUBLIC_HEADER mcqd.h)
+add_library(Mcqd::Mcqd ALIAS mcqd)
+export(TARGETS mcqd mcqdexe NAMESPACE Mcqd:: FILE McqdConfig.cmake)
+
+include(CMakePackageConfigHelpers) # exporting and installing ackage version
+write_basic_package_version_file(
+    McqdConfigVersion.cmake
+    VERSION ${PACKAGE_VERSION}
+    COMPATIBILITY AnyNewerVersion
+    )
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/McqdConfigVersion.cmake"
+        DESTINATION lib/cmake/Mcqd
+        )
+
+install(TARGETS mcqd mcqdexe # installing libs/execs/headers
+	EXPORT McqdConfig
+        LIBRARY DESTINATION lib
+        RUNTIME DESTINATION bin
+	PUBLIC_HEADER DESTINATION include
+        )
+install(EXPORT McqdConfig
+	FILE McqdConfig.cmake
+	NAMESPACE Mcqd::
+	DESTINATION lib/cmake/Mcqd
+        )
-- 
GitLab

