diff --git a/src/common.h b/src/common.h
index e7c10b8d0b..95e6f3d609 100644
--- a/src/common.h
+++ b/src/common.h
@@ -35,6 +35,10 @@
 // compiled with the wrong ABI
 GAP_STATIC_ASSERT(sizeof(void *) == SIZEOF_VOID_P, "sizeof(void *) is wrong");
 
+// if no GC is selected explicitly, default to GASMAN
+#if !defined(USE_BOEHM_GC) && !defined(USE_JULIA_GC) && !defined(USE_GASMAN)
+#define USE_GASMAN 1
+#endif
 
 // check for cygwin
 #if defined(__CYGWIN__) || defined(__CYGWIN32__)
