*** a/x86_64/Makefile.in	Mon Nov  4 14:08:05 2024
--- b/x86_64/Makefile.in	Mon Nov  4 14:15:46 2024
***************
*** 355,361 ****
  all: all-am
  
  .SUFFIXES:
! .SUFFIXES: .asm .lo .o .obj .s
  $(srcdir)/Makefile.in:  $(srcdir)/Makefile.am  $(am__configure_deps)
  	@for dep in $?; do \
  	  case '$(am__configure_deps)' in \
--- 355,361 ----
  all: all-am
  
  .SUFFIXES:
! .SUFFIXES: .asm .lo .o .obj .sx
  $(srcdir)/Makefile.in:  $(srcdir)/Makefile.am  $(am__configure_deps)
  	@for dep in $?; do \
  	  case '$(am__configure_deps)' in \
***************
*** 406,418 ****
  distclean-compile:
  	-rm -f *.tab.c
  
! .s.o:
  	$(AM_V_CCAS)$(CCASCOMPILE) -c -o $@ $<
  
! .s.obj:
  	$(AM_V_CCAS)$(CCASCOMPILE) -c -o $@ `$(CYGPATH_W) '$<'`
  
! .s.lo:
  	$(AM_V_CCAS)$(LTCCASCOMPILE) -c -o $@ $<
  
  mostlyclean-libtool:
--- 406,418 ----
  distclean-compile:
  	-rm -f *.tab.c
  
! .sx.o:
  	$(AM_V_CCAS)$(CCASCOMPILE) -c -o $@ $<
  
! .sx.obj:
  	$(AM_V_CCAS)$(CCASCOMPILE) -c -o $@ `$(CYGPATH_W) '$<'`
  
! .sx.lo:
  	$(AM_V_CCAS)$(LTCCASCOMPILE) -c -o $@ $<
  
  mostlyclean-libtool:
***************
*** 706,713 ****
  mulredc1_20.asm: mulredc1.m4
  	$(M4) -DLENGTH=20 $< > $@
  
! .asm.s:
! 	$(M4) -I../ -DOPERATION_$* `test -f $< || echo '$(srcdir)/'`$< >$*.s
  # Nothing here needs the C preprocessor, and including this rule causes
  # "make" to build .S, then .s files which fails on case-insensitive 
  # filesystems
--- 706,713 ----
  mulredc1_20.asm: mulredc1.m4
  	$(M4) -DLENGTH=20 $< > $@
  
! .asm.sx:
! 	$(M4) -I../ -DOPERATION_$* `test -f $< || echo '$(srcdir)/'`$< >$*.sx
  # Nothing here needs the C preprocessor, and including this rule causes
  # "make" to build .S, then .s files which fails on case-insensitive 
  # filesystems
