commit caa435144b1b4fa40c51558f15fd8dae2c0b5868
Author: Matthias Koeppe <mkoeppe@math.ucdavis.edu>
Date:   Sat Mar 11 10:16:56 2023 -0800

    BUG: handle ARCHFLAGS set to the empty string

diff --git a/mesonpy/__init__.py b/mesonpy/__init__.py
index e312059..d798f4c 100644
--- a/mesonpy/__init__.py
+++ b/mesonpy/__init__.py
@@ -686,8 +686,8 @@ class Project():
 
         # setuptools-like ARCHFLAGS environment variable support
         if sysconfig.get_platform().startswith('macosx-'):
-            archflags = self._env.get('ARCHFLAGS')
-            if archflags is not None:
+            archflags = self._env.get('ARCHFLAGS', '').strip()
+            if archflags:
                 arch, *other = filter(None, (x.strip() for x in archflags.split('-arch')))
                 if other:
                     raise ConfigError(f'Multi-architecture builds are not supported but $ARCHFLAGS={archflags!r}')
