diff --git a/src/maxima.in b/src/maxima.in
index ce84356a622..6af4c055238 100755
--- a/src/maxima.in
+++ b/src/maxima.in
@@ -28,7 +28,7 @@ setup_vars() {
       MAXIMA_IMAGESDIR="$MAXIMA_DEFAULT_IMAGESDIR"
   else
       if [ "$layout_autotools" = "true" ]; then
-         MAXIMA_IMAGESDIR=`unixize "$MAXIMA_PREFIX"`/lib/$PACKAGE/$MAXIMA_VERSION
+         MAXIMA_IMAGESDIR=`unixize "$MAXIMA_PREFIX"`/lib/x86_64-linux-gnu/$PACKAGE/$MAXIMA_VERSION
       else
          MAXIMA_IMAGESDIR=`unixize "$MAXIMA_PREFIX"`/src
       fi 
